### ISSUE TITLE
npm install hangs during install

### ISSUE THREAD

**Issue by simlal** [ValveSoftware/SteamOS#1186]
## System information

* Steam client version:
* SteamOS version: 3.4.10
* Opted into Steam client beta?: [Yes/No] No
* Opted into SteamOS beta?: [Yes/No]  No
* Have you checked for updates in Settings > System?: [Yes/No] No

## Issue
I had both `nix` and a debian image installed using `distrobox`  (with sudoless `podman`) on my deck. Had success using both ways to install packages, for example I got `Python` with `pip` and `venv` to install `scikit-learn`, `pandas` etc. inside a venv. I installed node (v18-x and v20-x) again with both ways (`nix` and inside the debian image) with success. Here's when it went awry:

I tried installing node modules with both `npm` and `yarn` and with both package managers I got stalled during the install for a random dependency (not the same on every reload). I tried clearing the npm/yarn cache, using http only in the configs but to no avail. With the verbose flag on npm I knew it was stuck on fetching a dependency at a given point in the installation. On yarn I got a `There appears to be trouble with your network connection. Retrying...` message. So far, what I've tried:
- using http in the npm/yarn config
- setting the `--network-timeout 100000` flag in npm/yarn
- I managed to install small packages such as `dotenv` with both npm/yarn
- I tried to disable IPV6, but did not work

### Steps for reproducing this issue:
**With nix**
1. `nix-shell -p nodejs-18_x` (or nodejs-20_x) 
2. `npm install puppeteer` (for example)
3. Example hangs on `⠹ reify:dependencyName: http fetch GET 200 https://registry.npmjs.org/dependencyName`
Tried with (`npm config set registry http://registry.npmjs.org/` or `npm set progress=false`, or `--network-timeout 100000`)
Same steps for `yarn` but it gave me the previously mentionned message `There appears to be trouble with your network connection. Retrying...`.

**Inside debian image**
Same steps as above but without activating the nix-shell just directly node 18 or 20x installed.

### What I haven't tried
- Using a wired connection
- Using another network or vpn
- With a VM inside SteamOS

*I know this feels like it would not belong in  a SteamOS issue and more of a npm/node thing, but it seems to have only happened in my case on the deck/SteamOS so this is why I am filing it out here!*

**Comment by Florin-Popescu** [comment]
Having the exact same issue for a few months now.

**Comment by bertogg** [comment]
I gave this a quick try and I can reproduce the problem easily with SteamOS 3.4 but not with 3.5 (I used nix and distrobox+debian in the latter case).

**Comment by simlal** [comment]
I might try later with a local install of nvm/node instead of Nix//Distrobox in 3.4 and see if the behavior is the same

**Comment by simlal** [comment]
so I did install nvm locally instead of with Nix. Went with node 20x, installed a small packaged and worked. Then tried TS and it stuck again. I cleared the cache, uninstalled version 20x, installed 18x (always via nvm). Installed ts/ts-node/dotenv and it worked!

I also got JavaSE20 and Maven running as well. I guess I'll stick with installing distributed bins and flatpaks for until 3.5 comes out

**Comment by tonivj5** [comment]
Same issue installing from n on SteamOS 3.4. I've tried every possible fix without success... (disable ipv6, firewall,...) 😢

**Comment by simlal** [comment]
from nix? I don't know about that but I haven't done a lot of js dev since but last time I used installed puppeteer via `npm install` and it worked.

Since 3.5, I saw the `/nix` folder and thus installed the nix package manager and successfully installed and used `bat`, `gcc`, `neofetch` and `vs-code` inside the base nix-env. Everything works great, and have been able to test the gcc compiler with a simple hello world and it worked (stoked now I can use my deck for my upcoming DSA class in C++). I even installed a specific Java SDK with the nix-shell for a temporary project and that worked great as well.

Since options to install locally or via nix worked fine I'll close the issue.

**Comment by simlal** [comment]
from nix? I don't know about that but I haven't done a lot of js dev since but last time I used installed puppeteer via `npm install` and it worked.

Since 3.5, I saw the `/nix` folder and thus installed the nix package manager and successfully installed and used `bat`, `gcc`, `neofetch` and `vs-code` inside the base nix-env. Everything works great, and have been able to test the gcc compiler with a simple hello world and it worked (stoked now I can use my deck for my upcoming DSA class in C++). I even installed a specific Java SDK with the nix-shell for a temporary project and that worked great as well.

Since options to install locally or via nix worked fine I'll close the issue.
